src/tox/session/env_select.py

```diff
@@ -241,13 +242,29 @@ def _finalize_config(self) -> None:
         self._state.conf.core.mark_finalized()
 
     def _build_run_env(self, name: str) -> RunToxEnv | None:
-        if self._provision is not None and self._provision[0] is False and name == self._provision[1]:
+        provision_on = provision_tox_env = provision_loader = None
```

Comment:
I think we just need to change the base for the provision env.

Reply:
not sure how to proceed on that one... but i'll think about it.

i'm off for the night here. it's been really fun hacking on `tox` today, thanks for the opportunity.

Reply:
See https://github.com/tox-dev/tox/blob/a2222e98b320467e821cf9444113e9b40ab3ad1b/src/tox/provision.py#L92; we should already not use inheritance from testenv here, we need to find out why it still happens. Likely a bug somewhere.

Reply:
Right in the middle of that now. I _think_ it's related to caching the `"base"` value from the Ini before the provision loader is even added to `env_conf`.

```
> /Users/masen/code/tox-dev/tox/.tox/py311/lib/python3.11/site-packages/tox/session/env_select.py(259)_build_run_env()
-> env_conf.loaders.insert(0, provision_loader)
(Pdb) env_conf._defined["base"]
ConfigDynamicDefinition(keys=('base',), desc=inherit missing keys from these sections, of_type=typing.List[str], default=['testenv'], _cache=['testenv'])
(Pdb) env_conf.loaders
[IniLoader(section=testenv, overrides={})]
```

If I reset the `_cache` values for the following keys after adding the `provision_loader` at index 0, then they can get the new values from the `provision_loader`.

```
(Pdb) tuple(env_conf._defined)
('set_env', 'setenv', 'base')
```

Something like

```
for config_def in env_conf._defined:
    if isinstance(config_def, ConfigDynamicDefinition):
        config_def._cache = _PLACE_HOLDER
```

But this actually doesn't work, because the loader for the `testenv` section is already in the chain at this point; removing it expicitly would be messy.

--------------

So the approach I'm taking now is to update `Config.get_env` to accept a `base` parameter, and if that's passed through it overrides the defaults. I couldn't find a way to reliably NOT include the base via config only, since shimming in the `provision_loader` occurs after `Config.get_env`, so the `testenv` default base just gets baked into the env's `loaders`.

Reply:
I think instead insert we need to use https://github.com/tox-dev/tox/blob/main/src/tox/session/cmd/devenv.py#L38, as it's too late for inserting the loader 🤔